File: guides/micronaut-jaxrs-jdbc/java/src/main/java/example/micronaut/Application.java
package example.micronaut;

import example.micronaut.domain.Owner;
import example.micronaut.domain.Pet;
import example.micronaut.domain.PetType;
import example.micronaut.repositories.OwnerRepository;
import example.micronaut.repositories.PetRepository;
import io.micronaut.context.event.ApplicationEventListener;
import io.micronaut.context.event.StartupEvent;
import io.micronaut.runtime.Micronaut;
import jakarta.inject.Singleton;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Arrays;

@Singleton // <1>
public class Application implements ApplicationEventListener<StartupEvent> { // <2>
    private static final Logger log = LoggerFactory.getLogger(Application.class);

    private final OwnerRepository ownerRepository;
    private final PetRepository petRepository;

    Application(OwnerRepository ownerRepository,
                PetRepository petRepository) { // <3>
        this.ownerRepository = ownerRepository;
        this.petRepository = petRepository;
    }

    public static void main(String[] args) {
        Micronaut.run(Application.class, args);
    }

    @Override
    public void onApplicationEvent(StartupEvent event) { // <2>
        log.info("Populating data");

        Owner fred = new Owner("Fred");
        fred.setAge(45);
        Owner barney = new Owner("Barney");
        barney.setAge(40);
        ownerRepository.saveAll(Arrays.asList(fred, barney)); // <4>

        Pet dino = new Pet("Dino", fred);
        Pet bp = new Pet("Baby Puss", fred);
        bp.setType(PetType.CAT);
        Pet hoppy = new Pet("Hoppy", barney);

        petRepository.saveAll(Arrays.asList(dino, bp, hoppy));
    }
}

File: guides/micronaut-database-authentication-provider/python/src/example/micronaut/application.py
from jakarta.inject import Singleton
from micronaut.runtime.server.event import ServerStartupEvent
from micronaut.runtime.event.annotation import EventListener

from .register_service import RegisterService


@Singleton
class Application:  # <1>
    def __init__(self, register_service: RegisterService):  # <2>
        self.register_service = register_service

    @EventListener
    def onApplicationEvent(self, event: ServerStartupEvent) -> None:  # <1>
        self.register_service.register(
            "sherlock@micronaut.example",
            "sherlock",
            "elementary",
            ["ROLE_DETECTIVE"],
        )  # <3>
